closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -120,7 +139,17 @@ protected void engineInit(Key key, SecureRandom random)
                         " are supported by the native implementation, " +
                         "using Java crypto implementation for key agreement.");
             }
-            this.initializeJavaImplementation(key, random);
+            this.initializeJavaImplementation(key);
+        }
+    }
+
+    @Override
+    protected void engineInit(Key key, SecureRandom random)
+            throws InvalidKeyException {
+        try {
+            init(key);
```

Comment:
Line 163 should use `init()` directly (like `ECDHKeyAgreement`):
```java
        init(key);
```

Reply:
Changed.